apps/client/src/features/table/store/CuesheetSettings.tsx

```diff
@@ -0,0 +1,55 @@
+import create from 'zustand';
+
+import { booleanFromLocalStorage } from '../../../common/utils/localStorage';
+
+interface CuesheetSettings {
+  showSettings: boolean;
+  followSelected: boolean;
+  showDelayBlock: boolean;
+  showPrevious: boolean;
+
+  toggleSettings: (val?: boolean) => void;
+  toggleFollow: (val?: boolean) => void;
+  toggleDelayVisibility: (val?: boolean) => void;
+  togglePreviousVisibility: (val?: boolean) => void;
+}
+
+function toggle(oldValue: boolean, value?: boolean) {
+  if (typeof value === 'undefined') {
+    return !oldValue;
+  }
+  return value;
+}
+
+enum CuesheetKeys {
+  Follow = 'ontime-cuesheet-follow-selected',
+  DelayVisibility = 'ontime-cuesheet-show-delay',
+  PreviousVisibility = 'ontime-cuesheet-show-previous',
+}
+
+export const useCuesheetSettings = create<CuesheetSettings>()((set) => ({
+  showSettings: false,
+  followSelected: booleanFromLocalStorage(CuesheetKeys.Follow, false),
+  showDelayBlock: booleanFromLocalStorage(CuesheetKeys.DelayVisibility, true),
+  showPrevious: booleanFromLocalStorage(CuesheetKeys.PreviousVisibility, true),
+
+  toggleSettings: (newValue?: boolean) => set((state) => ({ showSettings: toggle(state.showSettings, newValue) })),
+  toggleFollow: (newValue?: boolean) =>
+    set((state) => {
+      const value = toggle(state.followSelected, newValue);
```

Comment:
In principle I would agree. However since this code is so small, and will likely never grow beyond a few lines. I wonder if adding meaning to the variable would only add to the load on the reader

So in this case

```
  togglePreviousVisibility: (newValue?: boolean) =>
    set((state) => {
      const value = toggle(state.showPrevious, newValue);
      localStorage.setItem(CuesheetKeys.PreviousVisibility, String(value));
      return { showPrevious: value };
    }),

```

could become
```
  togglePreviousVisibility: (shouldShowPrevious?: boolean) =>
    set((state) => {
      const showPrevious = toggle(state.showPrevious, shouldShowPrevious);
      localStorage.setItem(CuesheetKeys.PreviousVisibility, String(showPrevious));
      return { showPrevious };
    }),

```

I wonder if `newValue` does more to signify to the reader "this is the thing the user wants to set" than a semantic variable name, such as in this example `shouldShowPrevious`) Although renaming the value to its name in the object does seem to improve readability

What do you think?

Reply:
Yeah, I don't mind if you take out the bad parts and use the good parts. Information overload is a thing and can be a problem!